cmd/bucket-handlers.go

```diff
@@ -715,7 +715,7 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 		return
 	}
 	if objectAPI.IsEncryptionSupported() {
-		if hasServerSideEncryptionHeader(formValues) && !hasSuffix(object, SlashSeparator) { // handle SSE-C and SSE-S3 requests
+		if crypto.IsRequested(formValues) && !hasSuffix(object, SlashSeparator) { // handle SSE requests
```

Comment:
validation needs to add a condition to reject sse-copy source headers if present, since crypto.IsRequested() is more generic now.
```
// X-Amz-Copy-Source shouldn't be set for this call.
	if _, ok := r.Header[xhttp.AmzCopySource]; ok {
		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrInvalidCopySource), r.URL, guessIsBrowserReq(r))
		return
	}
```

Reply:
Okay, but then this is a bug that existed before, right?! Since with the previous code we would not reject SSE-C copy headers either, right?!

Reply:
yes, looks like a preexisting bug here